modules/hostap/src/supp_api.c

```diff
@@ -75,6 +75,8 @@ static void supp_shell_connect_status(struct k_work *work);
 static K_WORK_DELAYABLE_DEFINE(wpa_supp_status_work,
 		supp_shell_connect_status);
 
+#define STR_CUR_TO_END(cur) (cur) = (&(cur)[0] + strlen((cur)))
```

Comment:
This causes the length to be calculated multiple times when this is called, why not have a len parameter in this macro and calculate the value only once.

Reply:
This macro define is not created in this PR. I just moved it to this location. This macro define was created and used for DPP feature.

Reply:
As I said, this macro calculates string length multiple times without any apparent benefit. You do not need to copy some macro from other place just because it exists there.
I will add more comments below.